src/reconcile/rec_visibility.c

```diff
@@ -617,14 +613,17 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *first_upd
             }
         }
 
-        /* Track the first update with non-zero timestamp. */
-        if (upd->start_ts > max_ts)
-            max_ts = upd->start_ts;
-
         /* Always select the newest committed update to write to disk */
         if (upd_select->upd == NULL)
             upd_select->upd = upd;
 
+        /* Track the selected update transaction id and timestamp. */
```

Comment:
I changed this code to track max transaction id and timestamp from the selected updates that are writing to the disk.

Reply:
I wonder if this defeats the other job of `rec_max_txn` which is to manage when content can be evicted to the WiredTiger metadata file. I don't have a concrete answer, but I'm nervous..

Reply:
No. This change shouldn't cause any problem with metadata page eviction. We will check the `page->mod->rec_max_txn` only when the page is clean. As we are leaving the page dirty because of newer updates on the page, this change doesn't cause any problems. Earlier also we track this information and we are not updating it on the page as the page is dirty.